src/store/juju/slice.ts

```diff
@@ -108,6 +123,31 @@ const slice = createSlice({
     clearControllerData: (state) => {
       state.controllers = {};
     },
+    fetchCrossModelQuery: (
+      state,
+      action: PayloadAction<CrossModelQueryRequest & WsControllerURLParam>
```

Comment:
I should have noticed this in my previous review, but I think this should just accept `query` as the `type` param is provided in the facade.

```suggestion
      action: PayloadAction<Pick<CrossModelQueryRequest["query"]> & WsControllerURLParam>
```

Reply:
I think it should be `Pick<CrossModelQueryRequest, "query">` instead. Also, due to this change, I think we also have to change `const { wsControllerURL, ...params } = action.payload;` to `const { wsControllerURL, query } = action.payload;` in `model-poller.ts`.

Reply:
Haha yes, that's how you use Pick, not how I wrote it :).